src/wh_client_crypto.c

```diff
@@ -1069,7 +1138,340 @@ int wh_Client_Curve25519SharedSecret(whClientContext* ctx,
 #endif
     return ret;
 }
-
 #endif /* HAVE_CURVE25519 */
 
+#ifndef NO_RSA
+int wh_Client_SetKeyIdRsa(RsaKey* key, whNvmId keyId)
+{
+    if (key == NULL)
+        return WH_ERROR_BADARGS;
+    key->devCtx = WH_KEYID_TO_DEVCTX(keyId);
+    return WH_ERROR_OK;
+}
+
+int wh_Client_GetKeyIdRsa(RsaKey* key, whNvmId* outId)
+{
+    if (key == NULL || outId == NULL)
+        return WH_ERROR_BADARGS;
+    *outId = WH_DEVCTX_TO_KEYID(key->devCtx);
+    return WH_ERROR_OK;
+}
+
+
+int wh_Client_ImportRsaKey(whClientContext* ctx, RsaKey* key,
+        whNvmFlags flags, uint32_t label_len, uint8_t* label,
+        whKeyId *out_keyId)
+{
+    int ret = 0;
+    whKeyId cacheKeyId = WH_KEYID_ERASED;
+    byte keyDer[WOLFHSM_CFG_COMM_DATA_LEN] = {0};
+    int derSize = 0;
+
+    if (    (ctx == NULL) ||
+            (key == NULL) ||
+            ((label_len != 0) && (label == NULL))) {
+        return WH_ERROR_BADARGS;
+    }
+
+    /* Convert RSA key to DER format */
+    ret = derSize = wc_RsaKeyToDer(key, keyDer, sizeof(keyDer));
```

Comment:
I don't believe this handles public keys - would we want a separate client function for public key import/export? Or we could just check `key->type` and call `wc_RsaKeyToPublicDer` if it is public? The only reason it is significant is it dictates which wolfCrypt API we use in order to generate the DER, as `wc_RsaKeyToDer` will fail if it is a public key.

Reply:
I saw this today and it should be changed to use the Serialize helper in wh_crypto.h that correctly calls the right export function based on the kind of rsa key.  I'll update and verify the functionality.  Probably need a few new test case to make sure all 3 cases work.